packages/api/blocks-api/src/blocks/block.ts

```diff
@@ -53,10 +53,16 @@ export declare type BlockIndexItem = {
 } & BlockIndexData;
 export declare type BlockIndex = Array<BlockIndexItem>;
 
+export interface BlockWarningReport {
+    message: string;
+    severity: string; // TODO: Use WarningSeverity here, as soon as it is in the package and not in the demo (cannot import from demo)
+}
+
 export interface BlockDataInterface {
     transformToPlain(context: BlockContext): Promise<Type<BlockTransformerServiceInterface> | TraversableTransformResponse>;
     transformToSave(): TraversableTransformResponse;
     indexData(): BlockIndexData;
+    reportWarnings(): BlockWarningReport[];
```

Comment:
Naming: I'd go with `warnings()` (like indexData and searchText) or `getWarnings()`. This doesn't really report the warnings, just collect them

Reply:
9fd3288b3667b7fa4a1f8bbafa9fb953a2143b34